File: nodejs/db/dbConnection.js
//! This file contains the connection to the database, and the functions to get a connection to the database.
const mysql = require("mysql");
require("dotenv").config();

const pool = mysql.createPool({
  connectionLimit: 10, //* Maximum number of connections to create at once
  host: process.env.DB_HOST,
  user: process.env.DB_USER,
  password: process.env.DB_PASSWORD,
  database: process.env.DB_NAME,
});

pool.on("connection", () => {
  console.log("\u001b[32m","Connected to the database.", "\u001b[0m");
  
});

const getConnection = (callback) => {
  pool.getConnection((err, connection) => {
    if (err) {
      console.error("Error connecting to database: ", err);
      return callback(err, null);
    }
    callback(null, connection);
  });
};

module.exports = {
  getConnection,
};

File: python/db/dbConnection.py
import mysql.connector
from mysql.connector import pooling
import os
from dotenv import load_dotenv

load_dotenv()

pool = pooling.MySQLConnectionPool(
    pool_name="mypool",
    pool_size=10,
    host=os.getenv("DB_HOST"),
    user=os.getenv("DB_USER"),
    password=os.getenv("DB_PASSWORD"),
    database=os.getenv("DB_NAME")
)

print("\u001b[32mConnected to the database.\u001b[0m")

def get_connection():
    try:
        connection = pool.get_connection()
        return connection
    except mysql.connector.Error as err:
        print("Error connecting to database: ", err)
        return None

if __name__ == "__main__":
    conn = get_connection()
    if conn:
        print("Successfully retrieved a database connection.")
        conn.close()